src/core/thread/mle_router.cpp

```diff
@@ -1808,7 +1809,6 @@ void MleRouter::HandleStateUpdateTimer(void)
             {
                 mRouters[i].mState = Neighbor::kStateInvalid;
                 mRouters[i].mLinkInfo.Clear();
-                mRouters[i].mNextHop = kInvalidRouterId;
```

Comment:
The primary purpose of this PR is to have `mNextHop` always maintain a path to the router through a neighboring router (if one exists).  The idea here is to invalidate the direct link but still allow routing through `mNextHop`.  I think the check in Line 1821 should still work since if there is still a route to the router.